### PR TITLE
tamatool: init at 0.1

### PR DESCRIPTION
**Pull request by fgaz**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/tamatool/default.nix [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+, copyDesktopItems
+, libpng
+, SDL2
+, SDL2_image
+
+# Optionally bundle a ROM file
+, rom ? null
+}:
+
+stdenv.mkDerivation rec {
```
Comment (drupol) [inline pkgs/applications/emulators/tamatool/default.nix:14]: I think we should use the `finalAttrs` pattern here.

Let me share a couple of useful links regarding this:

1. History: https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293
2. Documentation: https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes
3. Recent example of implementation: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe

Cheers.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/applications/emulators/tamatool/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/applications/emulators/tamatool/default.nix]: Done

**Review by drupol — APPROVED** [review]

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Dear @NixOS/darwin-maintainers, do you have a clue on the build failures on Darwin ?

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 234949` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by drupol** [comment]
Thanks @wegank !!!

**Pull request merged**